src/helpers.js

```diff
@@ -237,16 +237,16 @@ export function courseNumAsDecimal(courseNumber) {
 
     // This element is the second to last element of the array, supposedly a string of letters
     const secondToLastElement = filtered[filtered.length - 2];
-    // The character codes for [A-J] or [a-j] will turn into the character codes for 0-9, respectively
-    const charCode = lastElement.toUpperCase().charCodeAt(0) - 'A'.charCodeAt(0) + '0'.charCodeAt(0);
-    // Character codes 48 - 57 are the numbers 0 - 9
-    if ('0'.charCodeAt(0) - 1 < charCode && charCode < '9'.charCodeAt(0)) {
+    // The character codes for [A-J] or [a-j] will turn into 0-9, respectively
+    const charCode = lastElement.toUpperCase().charCodeAt(0) - 'A'.charCodeAt(0);
+    if (0 <= charCode && charCode <= 9) {
```

Comment:
reading this again, I realize that 9 actually should be excluded from the range since we're adding 1 afterward. Also, the variable `charCode` is now an inaccurate name since the number it represents isn't a character code anymore (that is to say, putting it into fromCharCode doesn't give you a sensical result).

Reply:
Since 39a -> 39.1, I assumed that ICS45J -> 45.10
However, for courseNumAsDecimal('ICS45J'), my code returns parseFloat('45.10'), which results in 45.1 being returned.
How should I address this problem? Should I return a string instead of numerical values?

Reply:
And would the name "letterAsNumber" work instead of "charCode"?

Reply:
45j should just return 45, since 10 is out of the range of one-digit numbers. And yeah that name works.

Reply:
I believe I have done it